### PR TITLE
Pin 4923: getPresignedUrl

### PR DESCRIPTION
**Pull request by AsterITA**
[To merge after #277 ]
https://pagopa.atlassian.net/browse/PIN-4923

### CODE REVIEW

File: src/main/resources/interface-specification.yml [hunk 1]
```diff
@@ -3876,6 +3876,53 @@ paths:
...
               schema:
                 $ref: '#/components/schemas/Problem'
+  /import/eservices/presignedUrl:
+    parameters:
+      - $ref: '#/components/parameters/CorrelationIdHeader'
+    get:
+      tags:
+        - eservices
+      summary: Get presigned URL
+      operationId: getPresignedUrl
```
Comment (galales) [inline src/main/resources/interface-specification.yml]:
```suggestion
      operationId: getImportEservicePresignedUrl
```

File: src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/EServicesApiServiceImpl.scala [hunk 2]
```diff
@@ -1040,6 +1040,25 @@ final case class EServicesApiServiceImpl(
...
+
+  override def getPresignedUrl()(implicit
+    contexts: Seq[(String, String)],
+    toEntityMarshallerPresignedUrl: ToEntityMarshaller[PresignedUrl],
+    toEntityMarshallerProblem: ToEntityMarshaller[Problem]
+  ): Route = {
+    val result: Future[PresignedUrl] = for {
+      presignedUrl <- fileManager.getPresignedUrl(
+        ApplicationConfiguration.importEServiceContainer,
+        ApplicationConfiguration.importEServicePath
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/EServicesApiServiceImpl.scala]: IMO the path should also contain the tenant id, it allows:
- easily debugging on users' reports
- implementing authorization rules

File: src/main/scala/it/pagopa/interop/backendforfrontend/common/system/ApplicationConfiguration.scala [hunk 3]
```diff
@@ -79,6 +79,8 @@ object ApplicationConfiguration {
   val consumerDocumentsPath: String      = config.getString("backend-for-frontend.storage.consumer-documents.path")
   val eServiceDocumentsContainer: String = config.getString("backend-for-frontend.storage.eservice-documents.container")
   val eServiceDocumentsPath: String      = config.getString("backend-for-frontend.storage.eservice-documents.path")
+  val importEServiceContainer: String     = config.getString("backend-for-frontend.storage.import-eservice.container")
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/backendforfrontend/common/system/ApplicationConfiguration.scala]:
```suggestion
  val importEServiceContainer: String    = config.getString("backend-for-frontend.storage.import-eservice.container")
```

File: src/main/scala/it/pagopa/interop/backendforfrontend/common/system/ApplicationConfiguration.scala
Comment (galales) [inline src/main/scala/it/pagopa/interop/backendforfrontend/common/system/ApplicationConfiguration.scala:1]: these new configurations should be added also to `application.conf` and `application-standalone.conf` files.

May I ask to open an IFST task to add these in the deployment?

File: src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/EServicesApiServiceImpl.scala [hunk 4]
```diff
@@ -1040,6 +1040,25 @@ final case class EServicesApiServiceImpl(
...
+      )
+    } yield presignedUrl
 
+    onComplete(result) {
+      val headers: List[HttpHeader] = headersFromContext()
+      handleError("Error getting presigned url", headers) orElse { case Success(resource) =>
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/EServicesApiServiceImpl.scala]:
```suggestion
      handleError("Error getting eservice import presigned url", headers) orElse { case Success(resource) =>
```

File: src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/EServicesApiServiceImpl.scala (hunk 2 above)
Comment (galales) [reply src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/EServicesApiServiceImpl.scala]: also, if I recall correctly, the presigned url request should also contain the file name

**Review by galales — CHANGES_REQUESTED** [review]

File: src/main/scala/it/pagopa/interop/backendforfrontend/common/system/ApplicationConfiguration.scala
Comment (AsterITA) [reply src/main/scala/it/pagopa/interop/backendforfrontend/common/system/ApplicationConfiguration.scala:1]: Task created: https://pagopa.atlassian.net/browse/PIN-4955

File: src/main/resources/interface-specification.yml [hunk 5]
```diff
@@ -3875,6 +3875,64 @@ paths:
             application/problem+json:
               schema:
                 $ref: '#/components/schemas/Problem'
+  /import/eservices/{tenantId}/presignedUrl:
```
Comment (galales) [inline src/main/resources/interface-specification.yml]: tenantId should be taken from the jwt, if we leave the caller to pass the value we lose the authorization check
```suggestion
  /import/eservices/presignedUrl:
```

**Review by galales — COMMENTED** [review]
just one more point 🙏

**Review by galales — APPROVED** [review]

**Review by beetlecrunch — APPROVED** [review]

**Pull request merged**